packages/aws-cdk/lib/util/directories.ts

```diff
@@ -9,4 +10,20 @@ export function cdkHomeDir() {
 
 export function cdkCacheDir() {
   return path.join(cdkHomeDir(), 'cache');
+}
+
+export function rootDir() {
+
+  function _rootDir(dirname: string): string {
+    const manifestPath = path.join(dirname, 'package.json');
+    if (fs.existsSync(manifestPath)) {
+      return dirname;
+    }
+    if (path.dirname(dirname) === dirname) {
+      throw new Error('Unable to find package manifest');
+    }
+    return _rootDir(path.dirname(dirname));
+  }
+
+  return _rootDir(__dirname);
```

Comment:
Note - in ESM mode, `__dirname` and `__filename` are not present...

Reply:
Yeah, we'll handle that when we switch over to ESM I suppose.